### PR TITLE
Fix 637/integrate intellij dub as submodule

### LINKED ISSUE #637: move the intellij-dub plugin code into a sub-module of this repo
The additional intellij-dub plugin was created before this codebase was split into modules. Having the separation in terms of code has been good but it's tricky maintaining it as a completely separate plugin with it's own releases and codebase. It also seems that recent changes in Intellij are making it harder to have intellij-dub rely on classes from the main repo.

Before putting the dub code into the project the following packages should probably be put into a core

```
io.github.intellij.dlanguage.icons.DlangIcons
io.github.intellij.dlanguage.module.DlangModuleType
```

These ones are already in *utils* so can either remain there or move into the new *core* module

```
io.github.intellij.dlanguage.messagebus.DubChangeNotifier
io.github.intellij.dlanguage.messagebus.ToolChangeListener
io.github.intellij.dlanguage.messagebus.Topics
io.github.intellij.dlanguage.settings.ToolKey
io.github.intellij.dlanguage.settings.ToolSettings
```

leaving these to either but put into the new *dub* module or reworked (ProcessDLibs could be rejigged)

```
io.github.intellij.dlanguage.project.DubConfigurationParser
io.github.intellij.dlanguage.project.DubPackage
io.github.intellij.dlanguage.run.DlangRunDubConfiguration
io.github.intellij.dlanguage.actions.ProcessDLibs
```

### PR DESCRIPTION
**Pull request by etienne02**
Closes #637 

Now all dub related code is in the dub submodule.

### CODE REVIEW

File: src/main/java/io/github/intellij/dlanguage/codeinsight/dcd/DCDCompletionServer.java [hunk 1]
```diff
@@ -166,7 +165,8 @@ GeneralCommandLine buildDcdCommand(@NotNull final String path) {
         });
 
         // try to auto add dub dependencies
-        final DubConfigurationParser dubParser = new DubConfigurationParser(module.getProject(),
+        // FIXME should not use dub to get dependencies
+        /*final DubConfigurationParser dubParser = new DubConfigurationParser(module.getProject(),
```
Comment (SingingBush) [inline src/main/java/io/github/intellij/dlanguage/codeinsight/dcd/DCDCompletionServer.java:169]: even though this should be changed probably best keeping it working as it is for this PR with a `// todo: ...` comment as commenting out the code will just break dcd completion

**Review by SingingBush — COMMENTED** [review]
looks great, I'm just a little concerned about breaking auto-completion

File: src/main/java/io/github/intellij/dlanguage/codeinsight/dcd/DCDCompletionServer.java (hunk 1 above)
Comment (etienne02) [reply src/main/java/io/github/intellij/dlanguage/codeinsight/dcd/DCDCompletionServer.java]: Solved, and improvements, now we don’t even need dub to get libraries for DCD command line

**Review by SingingBush — APPROVED** [review]
Awesome thanks

**Pull request merged**